File: meeting-1/foo.jl
@everywhere function single_fib(n::Int)
    _fib=single_fib
    n < 2 ? 1 : _fib(n-1) + _fib(n-2)
end

@everywhere function _parallel_fib(n::Int, level::Int)
    _fib(n::Int)= _parallel_fib(n, level+1)
    if n < 2
	1
    else
        # 2**6==64 already, but not all of them run long enough.
        if (level<11)
            a= @spawn _fib(n-1)
            b= _fib(n-2)
            fetch(a) + b
        else
            # _fib(n-1) + _fib(n-2)
            single_fib(n)
        end
    end
end

parallel_fib(n)= _parallel_fib(n, 0)

fib=parallel_fib

n = parse(Int,ARGS[1])
println(n)
println(fib(n))

File: meeting-1/foo.c
#include <assert.h>
#include <stdio.h>
#include <unistd.h>

typedef process_t pid_t;
// #define SPAWN(e) 
// or just:
process_t spawn ()

int64_t _fib (int64_t n, int level) {
    if (n < 2) {
	return 1;
    } else {
	if (level<6) {
	    process_t a= spawn(fib_, n-1);
	    int64_t b= fib_(n-2);
	    return FETCH(a) + b;
	} else {
	    return fib(n-1) + fib(n-2);
	}
    }
}

